src/OpenSage.Game/Logic/Object/GameObjectCollection.cs

```diff
@@ -56,7 +56,9 @@ public GameObject GetObjectById(int objectId)
         public void Remove(GameObject gameObject)
         {
             // TODO: We should probably be using an ID.
-            _items.Remove(gameObject);
+            var index = _items.IndexOf(gameObject);
```

Comment:
Not used

Reply:
Fixed that, we probably really shouldn't remove objects from the collection. They come in handy later on, for statistics purposes